uefi-raw/src/protocol/mod.rs

```diff
@@ -2,9 +2,25 @@
 
 //! Protocol definitions.
 //!
-//! Protocols are sets of related functionality identified by a unique
-//! ID. They can be implemented by a UEFI driver or occasionally by a
-//! UEFI application.
+//! # TL;DR
+//! Technically, a protocol is a `C` struct holding functions and/or data, with
+//! an associated [`GUID`].
+//!
+//! # About
+//! UEFI protocols are a structured collection of functions and/or data,
+//! identified by a [`GUID`], which defines an interface between components in
+//! the UEFI environment, such as between drivers, applications, or firmware
+//! services.
+//!
+//! Protocols are central to UEFI’s handle-based object model, and they provide
+//! a clean, extensible way for components to discover and use services from one
+//! another.
+//!
+//! Implementation-wise, a protocol is a `C` struct holding function pointers
+//! and/or data. Please note that some protocols may use [`core::ptr::null`] as
```

Comment:
@nicholasbishop I'd like to add more about protocol interfaces being null; why can they be null and when is this valid? Can you help?

Reply:
Here's the issue where it was first brought to our attention that the interface could be null: https://github.com/rust-osdev/uefi-rs/pull/859

So `LoadedImageDevicePath` on an image created with a null device path is one way, but probably not the only way to end up with a null interface.